pkg/image/prune/imagepruner.go

```diff
@@ -488,10 +557,33 @@ func (p *imagePruner) Run(pruneImage ImagePruneFunc, pruneStream ImageStreamPrun
 	allNodes := p.g.NodeList()
 
 	imageNodes := imageNodeSubgraph(allNodes)
+	if len(imageNodes) == 0 {
```

Comment:
When the ImagePruner is created via NewImagePruner, it calls addImagesToGraph, which only adds images managed by our registry.